### DISCUSSION TITLE
Bar gauge panel legend placement

### DISCUSSION THREAD

**Discussion by mxmorin** [grafana/grafana#37026] [Dashboards, Panels & Visualizations]
**What happened**:
When my query returns several lines, the legend is placed in front of the bars.
When it returns only one line (display name is valued at ${__series.name}), the legend is placed above the bars.

![image](https://user-images.githubusercontent.com/58942529/108242465-3c1ebd00-714d-11eb-84ac-3f3efee86149.png)

Is it possible to have the same placement whatever the return and better, provide a field to set it up ?

**What you expected to happen**:
Legend is place at the same place regardless of the number of lines returned or have a field to set the placement

**Environment**:
- Grafana version: 7.4.1
- Data source type & version: Prometheus
- OS Grafana is installed on: Centos 7
- User OS & Browser: Chrome

**Comment by dprokop** [comment]
@mxmorin thanks for this issue. The placement is calculated automatically based on the panel size, hence the differences.

I'm gonna classify this as a feature request for now and see if there is more interest in such a feature. In the meantime- we always welcome contributions from the community, and if you wanted to implement such a feature we are happy to review it!

**Comment by mxmorin** [comment]
Thanks for your response.
I think it should be better to add a panel section to deal with legends in the bar gauge panel. 
Differentiating the management of the legend according to the number of values (legend or displayName) as well as the calculation of the placement are rather obscure.
That's why I will suggeset to add a Show field with the values Always, Never, Multiple values only and another Placement with the values Above, In front, Below in order to manage all cases. 
Unfortunately, I don't have the necessary skills to develop this feature.

**Comment by cwallacetlc** [comment]
I have 5 different panels displaying bar gauges for similar data and different time ranges, all the same size (height & width).  2 of them show the labels beside the bars, and the other 3 show the labels above.  I need a way to make them consistent.

**Comment by alanmcg** [comment]
Layout features for the legend could also include placement on the right of the bars and / or text alignment could be configurable. Currently with label names in the legend it can look a bit strange with the right aligned text on the left of bars if there is a big variation in label length

**Comment by MikeyCarter** [comment]
Was there ever a resolution to this?    My series name is really long (as it's a name + primary key value which I use for selecting in other applications) so when it has multiple rows some of the key data is hidden by "..."

**Reply by leeoniya** [reply]
> Was there ever a resolution to this?

we're gonna try to put a timeline on a fix today.

> My series name is really long (as it's a name + primary key value which I use for selecting in other applications) so when it has multiple rows some of the key data is hidden by "..."

this is a separate discussion, but related to https://github.com/grafana/grafana/issues/75205

**Comment by nmarrs** [comment]
We are looking into addressing this in 10.2, please [refer to this github issue](https://github.com/grafana/grafana/issues/75928) to track progress. Thanks! We will close this discussion out in favor of continuing the conversation on the github issue / associated PRs.
